### ISSUE TITLE
Common API interface for waypoints

### ISSUE THREAD

**Issue by flixr** [paparazzi/paparazzi#981]
Currently rotorcraft and fixedwing firmwares have different waypoint representations/coordinate systems.
We have no common API interface to get the coordinates of a waypoint in one representation regardless if fw/rc firmware... needed e.g. for #936 

Also related to #455 

An ideas on what the best approach would be?
##

**Comment by flixr** [comment]
About the naming of the API functions for waypoints:
Currently we have `nav_set_waypoint_foo`, `nav_globalize_local_wp`, etc... in subsystems/navigation/waypoints.h
I think that was a bad initial choice on my part!
How about naming them all consistently (and shorter): `waypoint_set_enu_f`, `waypoint_set_xy_i`, `waypoint_get_lla`, `waypoint_localize`, `waypoints_globalize_all`, etc...?

**Comment by saaib** [comment]
Felix,

Is there a pre-agreed naming convenction structure for other functions/APIs?

Regards,

-Urivan

On 04/20/2015 05:33 AM, Felix Ruess wrote:
>
> About the naming of the API functions for waypoints:
> Currently we have |nav_set_waypoint_foo|, |nav_globalize_local_wp|, 
> etc... in subsystems/navigation/waypoints.h
> I think that was a bad initial choice on my part!
> How about naming them all consistently (and shorter): 
> |waypoint_set_enu_f|, |waypoint_set_xy_i|, |waypoint_get_lla|, 
> |waypoint_localize|, |waypoints_globalize_all|, etc...?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/paparazzi/paparazzi/issues/981#issuecomment-94440770>.
>

-- 
-Urivan Flores-Saaib

**Comment by flixr** [comment]
Not really, mostly for historic reasons... apart from what is in the style guide: http://docs.paparazziuav.org/latest/stylec.html

It is quite painful to change the API later, and this is not in any stable branch yet: so it should be decided before the next stable release.

In general functions and variables should start with a lower case letter.
And prefer under_scored_names, but we also have quite a few camelCase ones...
IMHO the most important thing, is to try to be consistent within the module/subsystem functional API.
